From c9e8a6efa5978e8177051f1781ebdc77ca3497fa Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 12 Dec 2024 13:33:50 +0800
Subject: [PATCH 282/301] AOSCOS: MIPS: lib: exclude generic memset.o if
 CPU_LOONGSON64 is set

In "AOSCOS: memset optimization for loongson-3", a Loongson-3-specific
memset implementation was introduced, with its own set of definitions
for `bzero' and `memset' routines. This resulted in multiple definitions
for the aforementioned assembly routines, causing vmlinux to fail to link.

Exclude generic memset.o if `CPU_LOONGSON64' is set to avoid this error.

Fixes: "AOSCOS: memset optimization for loongson-3"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/lib/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/lib/Makefile b/arch/mips/lib/Makefile
index 9c024e6d5e54..0b175c5ec9df 100644
--- a/arch/mips/lib/Makefile
+++ b/arch/mips/lib/Makefile
@@ -10,6 +10,7 @@ lib-y	+= bitops.o csum_partial.o delay.o memcpy.o memset.o \
 obj-y			+= iomap_copy.o
 obj-$(CONFIG_PCI)	+= iomap-pci.o
 lib-$(CONFIG_GENERIC_CSUM)	:= $(filter-out csum_partial.o, $(lib-y))
+lib-$(CONFIG_CPU_LOONGSON64)	:= $(filter-out memset.o, $(lib-y))
 
 obj-$(CONFIG_CPU_GENERIC_DUMP_TLB) += dump_tlb.o
 obj-$(CONFIG_CPU_R3000)		+= r3k_dump_tlb.o
-- 
2.48.1

